hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/Route.java

```diff
@@ -236,8 +238,32 @@ private MQQueueConnectionFactory mqQueueConnectionFactory()  {
         	mqQueueConnectionFactory.setPort(Integer.valueOf(properties.getValue(MQ_PORT)));
         	mqQueueConnectionFactory.setQueueManager(properties.getValue(MQ_QUEUEMANAGER));
     		logger.debug("{} - MQ connection factory has been created.", methodName);			
-        } catch (JMSException jmse) {
-        	 logger.error("{} - MQ connection factory initialization failed with the error : {}", methodName, jmse.getMessage());       	
+        } catch (JMSException je) {
+        	// Check for linked exceptions in JMSException
+            Throwable t = je;
+            while (t != null) {          
```

Comment:
There's some utilities in org.apache.commons.lang3.exception.ExceptionUtils that might simplify the code. For example getThrowableList() will return a list of all Throwables that you could forEach over instead of using a while loop.